orc/src/test/java/org/apache/iceberg/orc/TestOrcDataWriter.java

```diff
@@ -126,4 +135,125 @@ public void testDataWriter() throws IOException {
 
     Assert.assertEquals("Written records should match", records, writtenRecords);
   }
+
+  @Test
+  public void testNoSupportForDummyScheme() throws IOException {
+    ProxyOutputFile outFile = new ProxyOutputFile(Files.localOutput(temp.newFile()));
+    Assertions.assertThrows(
+        RuntimeIOException.class,
+        () -> HadoopOutputFile.fromPath(new Path(outFile.location()), new Configuration()),
+        "No FileSystem for scheme \"dummy\"");
+  }
+
+  @Test
+  public void testUsingFileIO() throws IOException {
+    // Show that FileSystem access is not possible for the file we are supplying as the scheme
+    // dummy is not handled
+    ProxyOutputFile outFile = new ProxyOutputFile(Files.localOutput(temp.newFile()));
+    Assertions.assertThrows(
```

Comment:
it would be better to move such assertions to AssertJ, because otherwise it'll be more work potentially to move such code when upgrading to Junit5

Reply:
Makes sense. Changing.